formula/src/main/java/com/instacart/formula/internal/ActionManager.kt

```diff
@@ -110,43 +112,25 @@ internal class ActionManager(
     }
 
     private fun prepareNewActionList() {
-        if (!startListInvalidated) {
-            return
-        }
+        if (recomputeCheckToStartList) {
+            recomputeCheckToStartList = false
 
-        startListInvalidated = false
-        scheduledToStart?.clear()
+            checkToStartActionList?.clear()
 
-        val actionList = actions ?: emptyList()
-        if (!actionList.isEmpty()) {
-            if (scheduledToStart == null) {
-                scheduledToStart = mutableListOf()
-            }
-            scheduledToStart?.addAll(actionList)
-            for (action in actionList) {
-                if (!isRunning(action)) {
```

Comment:
This logic was poorly written as we were potentially adding to the list twice and checking `isRunning()` multiple times.